r/src/array.cpp

```diff
@@ -220,11 +234,19 @@ std::shared_ptr<arrow::Array> Array__from_vector(SEXP x) {
       return arrow::r::MakeBooleanArray(x);
     case INTSXP:
       if (Rf_isFactor(x)) {
-        break;
+        return arrow::r::MakeFactorArray(x);
+      }
+      if (Rf_inherits(x, "Date")) {
+        return arrow::r::SimpleArray<INTSXP, arrow::Date32Type>(x);
       }
       return arrow::r::SimpleArray<INTSXP, arrow::Int32Type>(x);
     case REALSXP:
-      // TODO: Dates, ...
+      if (Rf_inherits(x, "Date")) {
+        return arrow::r::SimpleArray<INTSXP, arrow::Date32Type>(x);
+      }
+      if (Rf_inherits(x, "POSIXct")) {
+        return arrow::r::SimpleArray<REALSXP, arrow::Date64Type>(x);
+      }
```

Comment:
These have the same representation?

Reply:
I’ll double check. No that you say this, R representation might be seconds since epoch

Reply:
So we're fine (if I understand this correctly) between these two dates: 

```r
> structure(c(-2^52, 2^52), class = c("POSIXct", "POSIXt"))
[1] "-142711421-01-25 20:21:05 LMT" "142715360-12-06 04:48:16 CET"
```